salt/netapi/rest_cherrypy/app.py

```diff
@@ -1422,6 +1443,26 @@ def POST(self, **kwargs):
         else:
             creds = cherrypy.serving.request.lowstate
 
+        # Check acl for the api.
+        failure_str = ("[api_acl] Authentication failed for "
+                       "user {0} from IP {1}")
+        success_str = ("[api_acl] Authentication sucessful for "
+                       "user {0} from IP {1}")
+        user = creds.get('username', '')
+
+        # Check if handling a request from a proxy.
+        ip = cherrypy.request.headers.get('X-Forwarded-For', None)
```

Comment:
I'm still working on this, my weekend got away from me and I'll update this shortly.  Thanks.